### ISSUE TITLE
How does Post-Processor 'vsphere' and 'vsphere-template' work?

### ISSUE THREAD

**Issue by matteoiacovitti** [hashicorp/packer#9331]
Hello,

I created a template for packer to create a VM and install Fedora CoreOS on it. Once the VM is prepared, I run some post-processors that includes a script to add a custom template to the VM and convert the off files to an ova.

However, before that happens, I want to run the vsphere and vsphere-template processors so that I can have a copy in my vCenter.

My current config  for this is:

```
"post-processors": [
    {
      "type": "shell-local",
      "scripts": [
        "files/scripts/xml-ip-ovf.sh"
      ]
    },
    [
      {
        "type": "vsphere",
        "host":       "{{user `vServer`}}",
        "username":   "{{user `vUser`}}",
        "password":   "{{user `vPass`}}",
        "insecure":   "true",
        "datacenter": "{{user `vDatacenter`}}",
        "cluster":    "{{user `vCluster`}}",
        "vm_name":    "ASAA_vCenter_{{user `version`}}",
        "vm_folder":  "{{user `vFolder`}}",

        "keep_input_artifact": true,
        "overwrite":  true
      },
      {
        "type": "vsphere-template",
        "host":       "{{user `vServer`}}",
        "username":   "{{user `vUser`}}",
        "password":   "{{user `vPass`}}",
        "insecure":   "true",
        "datacenter": "{{user `vDatacenter`}}",
        "folder":  "/{{user `vFolder`}}",
        "keep_input_artifact": true
      }
    ],
    {
      "type": "shell-local",
      "scripts": [
      "files/scripts/new-ovf-to-ova.sh"
      ]
    }
  ]
```
When Post-processor kicks in, I get this error after packer completes

```
* Post-processor failed: Error uploading virtual machine: exit status 1


2020/05/29 23:52:35 machine readable: error-count []string{"1"}
==> Some builds didn't complete successfully and had errors:
2020/05/29 23:52:35 machine readable: vsphere-iso,error []string{"1 error(s) occurred:\n\n* Post-processor failed: Error uploading virtual machine: exit status 1\n\n"}
Build 'vsphere-iso' errored: 1 error(s) occurred:

* Post-processor failed: Error uploading virtual machine: exit status 1


==> Builds finished but no artifacts were created.
* Post-processor failed: Error uploading virtual machine: exit status 1



2020/05/29 23:52:35 [INFO] (telemetry) Finalizing.
==> Some builds didn't complete successfully and had errors:
--> vsphere-iso: 1 error(s) occurred:

* Post-processor failed: Error uploading virtual machine: exit status 1
```
I read the documentation and I don't see what I am doing wrong. 
Is there a step I'm missing?

**Comment by nywilken** [comment]
Hi @matteoiacovitti it's a little hard for me to make sense of what might be going on just by looking at the provided logs. Is it possible to provide the full debug logs `PACKER_LOG=1 packer build template.json`? 

Also you mention that you have a Packer template to create a VM, but I don't see any builders in your configuration. Are you chaining builds (template to build vm, then this template to run post processors)?

**Comment by matteoiacovitti** [comment]
Sorry, @nywilken for the confusion. I only posted the Post-Processor part of the template since that was the problem. Also, the log I posted was with PACKER_LOG enabled.
Over the weekend, however, I figured out what was wrong.

I have 2 custom scripts that run to add a custom template in the OVA and then convert the off files to an OVA. But I put both vsphere and vsphere-template processors before a script. I moved both scripts to the top and then executed the vsphere and vsphere-template processors and it worked. I think the scripts were messing about with the artifacts that the builder created.

Thanks.
